packages/core/nuxt-module/lib/plugins/i18n-cookies.js

```diff
@@ -69,20 +70,24 @@ const i18nCookiesPlugin = ({ $cookies, i18n, app, redirect }) => {
   if (missingFields.length) {
     throw new Error(`Following fields are missing in the i18n configuration: ${missingFields.join(', ')}`);
   }
-
-  if (cookieLocale !== settings.locale) {
+  if (cookieLocale !== settings.locale && !!i18nOptions.autoChangeCookie?.locale || !$cookies.get(cookieNames.locale)) {
```

Comment:
I think it would be better not to check every time if `autoChangeCookie` exists. Maybe extend `settings` variable with those options for `autoChangeCookie`  with providing default value. Default should be `true` as it is current default behavior.